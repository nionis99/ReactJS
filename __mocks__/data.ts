export const genres = ['Crime', 'Documentary', 'Horror', 'Comedy'];
export const MovieMockData = {
  id: 0,
  title: 'The Avengers 2',
  vote_average: 7.5,
  release_date: '2012-04-25',
  poster_path: 'https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
  genres: ['Science Fiction', 'Action', 'Adventure'],
  runtime: 143,
};
