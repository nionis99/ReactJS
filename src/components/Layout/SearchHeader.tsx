import React, { HTMLAttributes, useState } from 'react';
import AppLogo from '../Logo';
import Button from '../Button';

const SearchHeader = ({ className = '', ...rest }: HTMLAttributes<HTMLDivElement>) => {
  const [searchValue, setSearchValue] = useState('');

  const onSubmit = () => {
    if (searchValue) alert(`Looking for: ${searchValue}`);
  };

  return (
    <header className={`flex-col h-96 bg-header py-5 px-16 text-white bg-cover ${className}`} {...rest}>
      <div className="flex w-full justify-between items-start">
        <AppLogo />
        <Button title="+ ADD MOVIE" variant="secondary" size="small" />
      </div>
      <div className="flex-col mx-16 my-11">
        <label className="flex text-5xl">FIND YOUR MOVIE</label>
        <div className="flex mt-9 ">
          <input
            className="flex rounded p-4 text-lg w-full bg-gray80 opacity-60"
            name="search"
            onInput={(e) => setSearchValue(e.currentTarget.value)}
            placeholder="What do you want to watch?"
          />
          <Button
            className="flex ml-2 transform uppercase"
            title="Search"
            variant="primary"
            size="large"
            onClick={onSubmit}
          />
        </div>
      </div>
    </header>
  );
};

export default SearchHeader;
