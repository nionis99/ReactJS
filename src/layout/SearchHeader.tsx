import React, { HTMLAttributes, useState } from 'react';
import AppLogo from 'components/Logo';
import Button from 'components/Button';
import MovieFormModal from 'components/Modals/MovieFormModal';

const SearchHeader = ({ className = '', ...rest }: HTMLAttributes<HTMLDivElement>) => {
  const [searchValue, setSearchValue] = useState('');
  const [isMovieFormModalOpen, setIsMovieFormModalOpen] = useState(false);

  const onSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    if (searchValue) alert(`Looking for: ${searchValue}`);
  };

  const openAddMovie = () => setIsMovieFormModalOpen(true);
  const closeAddMovie = () => setIsMovieFormModalOpen(false);

  return (
    <header className={`flex-col h-96 bg-header py-5 px-16 text-white bg-cover ${className}`} {...rest}>
      <MovieFormModal isOpen={isMovieFormModalOpen} onClose={closeAddMovie} title="Add movie" />
      <div className="flex w-full justify-between items-start">
        <AppLogo />
        <Button title="+ ADD MOVIE" variant="secondary" size="small" onClick={openAddMovie} />
      </div>
      <div className="flex-col mx-16 my-11">
        <label className="flex text-5xl">FIND YOUR MOVIE</label>
        <form className="flex mt-9">
          <input
            className="flex rounded p-4 text-lg w-full bg-gray80 opacity-60 focus:outline-none"
            name="search"
            onInput={(e) => setSearchValue(e.currentTarget.value)}
            placeholder="What do you want to watch?"
          />
          <Button
            className="flex ml-2 uppercase"
            title="Search"
            variant="primary"
            type="submit"
            size="large"
            onClick={onSubmit}
          />
        </form>
      </div>
    </header>
  );
};

export default SearchHeader;
