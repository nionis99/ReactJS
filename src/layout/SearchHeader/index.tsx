import React, { HTMLAttributes } from 'react';
import AppLogo from 'components/Logo';
import Button from 'components/Button';
import constants, { testingConstants } from 'utils/Constants';

interface SearchHeaderProps extends HTMLAttributes<HTMLDivElement> {
  defaultSearchValue: string;
  setSearchValue: (searchValue: string) => void;
  onSearchSubmit: (e: React.MouseEvent<HTMLButtonElement>) => void;
  openAddMovie: () => void;
}

const SearchHeader = ({
  defaultSearchValue,
  setSearchValue,
  onSearchSubmit,
  openAddMovie,
  className = '',
  ...rest
}: SearchHeaderProps) => (
  <header className={`flex-col h-96 bg-header py-5 px-4 md:px-16 text-white bg-cover ${className}`} {...rest}>
    <div className="flex w-full justify-between items-start">
      <AppLogo />
      <Button
        buttonTitle="+ ADD MOVIE"
        variant="secondary"
        size="small"
        onClick={openAddMovie}
        data-testid={testingConstants.addMovieButton}
      />
    </div>
    <div className="flex-col m-2 md:mx-16 md:my-11">
      <label className="flex text-2xl md:text-5xl">{constants.headerTitle}</label>
      <form className="flex mt-2 md:mt-9 w-full">
        <input
          className="flex rounded p-1 md:p-4 text-lg w-full bg-gray80 opacity-60 focus:outline-none"
          name="search"
          onInput={(e) => setSearchValue(e.currentTarget.value)}
          placeholder="What do you want to watch?"
          defaultValue={defaultSearchValue}
          data-testid={testingConstants.searchHeaderInput}
        />
        <Button
          className="flex ml-2 uppercase"
          buttonTitle="Search"
          variant="primary"
          size="large"
          onClick={onSearchSubmit}
          data-testid={testingConstants.searchSubmitButton}
        />
      </form>
    </div>
  </header>
);

export default SearchHeader;
