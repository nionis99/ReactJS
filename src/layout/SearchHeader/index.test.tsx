import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render } from '@testing-library/react';
import { MovieMockData } from '../../../__mocks__/data';
import constants, { testingConstants } from 'utils/Constants';
import SearchHeader from '.';

describe('Movie search header component', () => {
  const onClickAddMovie = jest.fn();
  const onSearchSubmit = jest.fn();
  const setSearchValue = jest.fn();
  const wrapper = (
    <BrowserRouter>
      <SearchHeader
        onSearchSubmit={onSearchSubmit}
        setSearchValue={setSearchValue}
        defaultSearchValue={MovieMockData.title}
        openAddMovie={onClickAddMovie}
      />
    </BrowserRouter>
  );

  it('should render movie search header component', () => {
    const elem = renderer.create(wrapper).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should call onSubmit function', () => {
    const { getByText, getByTestId } = render(wrapper);
    expect.assertions(3);
    expect(getByText(constants.headerTitle)).toBeInTheDocument();
    fireEvent.change(getByTestId(testingConstants.searchHeaderInput), { target: { value: MovieMockData.title } });
    expect(getByTestId(testingConstants.searchHeaderInput)).toHaveValue(MovieMockData.title);
    fireEvent.click(getByTestId(testingConstants.searchSubmitButton));
    expect(onSearchSubmit).toBeCalledTimes(1);
  });
});
