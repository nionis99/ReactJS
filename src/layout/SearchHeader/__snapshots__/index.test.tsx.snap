// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie search header component should render movie search header component 1`] = `
<header
  className="flex-col h-96 bg-header py-5 px-4 md:px-16 text-white bg-cover "
>
  <div
    className="flex w-full justify-between items-start"
  >
    <test-file-stub
      height={24}
      width={150}
    />
    <button
      className="flex justify-center items-center rounded border p-2 text-primary bg-gray68 border-gray68 px-1 py-0.5 md:px-4 md:py-1"
      data-testid={0}
      onClick={[MockFunction]}
    >
      + ADD MOVIE
    </button>
  </div>
  <div
    className="flex-col m-2 md:mx-16 md:my-11"
  >
    <label
      className="flex text-2xl md:text-5xl"
    >
      FIND YOUR MOVIE
    </label>
    <form
      className="flex mt-2 md:mt-9 w-full"
    >
      <input
        className="flex rounded p-1 md:p-4 text-lg w-full bg-gray80 opacity-60 focus:outline-none"
        data-testid={1}
        defaultValue="The Avengers 2"
        name="search"
        onInput={[Function]}
        placeholder="What do you want to watch?"
      />
      <button
        className="flex justify-center items-center rounded border p-2 text-white bg-primary border-primary px-4 py-2 md:px-16 md:py-4 flex ml-2 uppercase"
        data-testid={2}
        onClick={[MockFunction]}
      >
        Search
      </button>
    </form>
  </div>
</header>
`;
